test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -418,10 +421,10 @@ func runWorkload(testCtx context.Context, b *testing.B, tc *testCase, w *workloa
 		// be complete.
 		b.Fatal(err)
 	}
-	for opIndex, err := range barrierAndCollectErrors {
-		if err != nil {
-			b.Fatalf("op %d: %v", opIndex, err)
-		}
+	select {
+	case err := <-opErrorChan:
+		b.Fatalf("op %d: %v", err.opIndex, err.err)
+	default:
 	}
 	return dataItems
```

Comment:
there are no guarantees here that there is no `barrierAndCollect` still running at this point. Maybe we can add a wait group.